docker/main/ngen/Dockerfile

```
@@ -374,32 +394,37 @@ RUN echo "export HYDRA_HOST_FILE=${HYDRA_HOST_FILE}" >> /etc/profile \
     && chown -R ${USER}:${USER} ${BOOST_ROOT} \
     && rm -rf /tmp/ngen-deps
 
-# Install required Python packages for troute build/run that needed one or more of the source dependencies
-RUN pip install --no-cache-dir numpy pandas pyyaml bmipy Cython blosc2==${BLOSC2_VERSION:-2.0.0} netCDF4 wheel packaging \
-    && env HDF5_DIR=/usr pip install --no-cache-dir --no-build-isolation tables
-
 ENV PATH=${PATH}:/usr/lib64/mpich/bin
 ENV LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:/usr/lib:/usr/local/lib64
+# This next value is eventually needed for the t-route build ...
+# ... make sure it stays in sync with configure step for netcdf above
+ENV NETCDFINC=/usr/include
+ENV CC=gcc
+
+# Get this for the times we need to install t-route
+COPY --from=rocky_init_troute_repo ${WORKDIR}/t-route/requirements.txt ${WORKDIR}/t-route-requirements.txt
+
+# Install troute and required Python packages, with some deps needing some special attention
+RUN if [ "${NGEN_WITH_ROUTING}" == "ON" ]; then \
+      # Get subset of requirements needed first (including version specifiers if present) because of netCDF4 handling \
+      cat ${WORKDIR}/t-route-requirements.txt | grep -i numpy >> first_reqs.txt ; \
+      cat ${WORKDIR}/t-route-requirements.txt | grep -i pandas | grep -v geo >> first_reqs.txt ; \
+      cat ${WORKDIR}/t-route-requirements.txt | grep -i pyyaml >> first_reqs.txt ; \
+      cat ${WORKDIR}/t-route-requirements.txt | grep -i cython >> first_reqs.txt ; \
+      cat ${WORKDIR}/t-route-requirements.txt | grep -i pyarrow >> first_reqs.txt ; \
+      pip3 install --no-cache-dir -r first_reqs.txt ; \
+      rm first_reqs.txt ; \
+      # Then install a few more things, including a particular blosc2 version and an adjusted build of tables \
```

> Looks like [`fiona`](https://pypi.org/project/fiona/1.9.5/) is the culprit. `geopandas` depends on `fiona` and `fiona` by default will install from `pip` with a pre-built `libgdal`, _if it can_. It seems like the issue is `fiona` does not have a distribution with pre-build `libgdal` for the particular target.
> 
> > Fiona has several [extension modules](https://docs.python.org/3/extending/extending.html) which link against libgdal. This complicates installation. Binary distributions (wheels) containing libgdal and its own dependencies are available from the Python Package Index and can be installed using pip.

[source](https://github.com/NOAA-OWP/DMOD/issues/475#issuecomment-1883705630)

```suggestion
      # Then install a few more things, including a particular blosc2 version and an adjusted build of tables \
      dnf install -y gdal-devel ; \
```

---

I fixed this differently, by including `gdal-devel` in the main step where the list of packages to be installed via `dnf` is defined.  It'll be installed now in an ancestor build stage of the one the suggestion would have been in.